developer-process/publish-checklist.md

```diff
@@ -0,0 +1,122 @@
+
+3. If there were several conflicts in cherry-pick process and you have to make several changes in process of selecting partial changes from a PR, create a PR, get teams review, otherwise just push all changes into release branch directly.
+
+4. Test the branch by running `npm run test` and website examples.
+
+5. Update CHANGELOG.md, making sure all commits and PRs merged after release are recorded properly:
+<div align="center">
+  <div>
+    <img src="https://raw.github.com/visgl/deck.gl-data/master/images/dev-docs/publish-guideline/image4.png" />
+    <p><i>Image Text</i></p>
+  </div>
+</div>
+
+6. Make sure you are using the correct NPM profile, then run the publish script:
+  ```bash
+  npmrc public
+  git add .
+  npm run publish prod
+  ```
+  Double check the version numbers before confirming to publish.
+
+
+## Beta release
+
+1. Switch to the master branch.
+  ```bash
+  git checkout master
+  git pull
+  ```
+
+2. Test the branch by running `npm run test` and website examples and layer-browser.
+
+3. Update CHANGELOG.md.
+
+4. Only if this is the first pre-release of a new version: open `lerna.json`, change the `version` field to `<version>-alpha.0` or `<version>-beta.0`.
+
+5. Make sure you are using the correct NPM profile, then run the publish script:
+  ```bash
+  npmrc public
+  git add .
+  npm run publish beta
+  ```
+  Double check the version numbers before confirming to publish.
+
+
+## Major/Minor release
+
+
+### Cut the release branch
+
+1. The latest release branch should be created by duplicating the master branch:
+  ```bash
+  git checkout master
+  git pull
+  git checkout -b <X.x>-release
+  ```
+2. Update the links in documentation and website to point to the current branch:
+  ```bash
+  npm run update-release-branch <X.x>
+  ```
+3. Push to upstream:
+  ```bash
+  git push
+  ```
+
+### Build the website
+
+4. Under the release branch:
+ ```bash
+ cd website
+ yarn
+ yarn build
+ ```
+
+### Pre-release checks
+
+5. The files inside `website/dist` or `website/public` are the production build of the website. Stage the website on a static server to test in all supported OS and browsers.
+
+6. Test the branch by running `npm run test` and website examples.
+
+7. Check all the bug/feature tickets under the current github milestone. Make sure they are properly listed in What's New (`/docs/whats-new.md`) and Upgrade Guide (`/docs/upgrade-guide.md`). Open a PR to update these pages.
+
+8. Update CHANGELOG.md.
+
+9. Make sure you are using the correct NPM profile, then run the publish script:
+  ```bash
+  npmrc public
+  git add .
+  npm run publish prod
+  ```
+  Double check the version numbers before confirming to publish.
+
+
+### Publish the website
+
+* Copy the content of `website/dist` or `website/public` folder to a different place
+* Now checkout `gh-pages` branch (`git checkout gh-pages`)
```

Comment:
I am happy to say that we have mostly switched to just using `yarn deploy` in the websites now`